### PR TITLE
CB-7938: Enable delete protection for RDS to avoid accidental deletions.

### PR DESCRIPTION
**Pull request by kkalvagadda1**
**Solution approach:**

When the RDS is created we should enable delete protection. This makes sure that customers don't accidentally delete the database. 

While terminating the database, we need to disable the delete protection if enabled, and then terminate the database.

### CODE REVIEW

File: cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsRdsStatusLookupService.java [hunk 1]
```diff
@@ -44,6 +49,28 @@ public ExternalDatabaseStatus getStatus(AuthenticatedContext ac, DatabaseStack d
                 .get();
     }
 
+    public boolean isDeleteProtectonEnabled(AuthenticatedContext ac, DatabaseStack dbStack) {
```
Comment (doktoric) [inline cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsRdsStatusLookupService.java:52]: isDeleteProtectionEnabled

File: cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsResourceConnector.java [hunk 2]
```diff
@@ -118,6 +121,10 @@ private boolean deployingToSameVPC(AwsNetworkView awsNetworkView, boolean existi
     @Override
     public List<CloudResourceStatus> terminateDatabaseServer(AuthenticatedContext ac, DatabaseStack stack,
             List<CloudResource> resources, PersistenceNotifier persistenceNotifier, boolean force) throws Exception {
+        if(awsRdsStatusLookupService.isDeleteProtectonEnabled(ac, stack)) {
```
Comment (doktoric) [inline cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsResourceConnector.java:124]: isDeleteProtectionEnabled

File: cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/scheduler/CustomAmazonWaiterProvider.java [hunk 3]
```diff
@@ -81,4 +81,37 @@ public WaiterState getState() {
...
                 .withExecutorService(WaiterExecutorServiceFactory.buildExecutorServiceForWaiter("AmazonRDSWaiters")).build();
     }
+
+    public Waiter<DescribeDBInstancesRequest> getDbInstanceModifyWaiter(AmazonRDS rdsClient) {
+        return new WaiterBuilder<DescribeDBInstancesRequest, DescribeDBInstancesResult>()
+                .withSdkFunction(new DescribeDBInstancesFunction(rdsClient))
+                .withAcceptors(new WaiterAcceptor<DescribeDBInstancesResult>() {
+                    @Override
+                    public boolean matches(DescribeDBInstancesResult describeDBInstancesResult) {
+                        return describeDBInstancesResult.getDBInstances().stream().allMatch(instance -> "avaialble".equals(instance.getDBInstanceStatus()));
```
Comment (doktoric) [inline cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/scheduler/CustomAmazonWaiterProvider.java:91]: available ?

**Review by doktoric — APPROVED** [review]

File: cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsResourceConnector.java (hunk 2 above)
Comment (kkalvagadda1) [reply cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsResourceConnector.java]: fixed.

File: cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsRdsStatusLookupService.java (hunk 1 above)
Comment (kkalvagadda1) [reply cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsRdsStatusLookupService.java]: fixed.

File: cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/scheduler/CustomAmazonWaiterProvider.java (hunk 3 above)
Comment (kkalvagadda1) [reply cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/scheduler/CustomAmazonWaiterProvider.java:91]: yes. Updated.

**Pull request merged**